server/graphql/v2/interface/Account.js

```diff
@@ -463,11 +463,19 @@ const accountFieldsDefinition = () => ({
         merchantId = (await fetchAccountWithReference(args.merchantAccount, { throwIfMissing: true })).id;
       }
 
+      let where;
+      if (account.type === CollectiveTypes.USER) {
+        const user = await models.User.findOne({
+          where: { CollectiveId: account.id },
+        });
+        where = { UserId: user.id };
```

Comment:
`UserId` is meant to store the user who created the card. It seems like a hack to re-use `account` here, and I'm not sure how it will play if we ever roll out virtual cards for other hosts (whose type can be `USER`).

I'd be in favor of an `assignee` (Individual) argument to achieve this.

Reply:
> no_entry_sign `UserId` is meant to store the user who created the card. 

I am not sure what you mean by this. To me it seems that the `UserId` within the `VirtualCard` table stores the user that the virtual card is assigned. In fact we use `UserId` to computer the `assignee` field here; 

https://github.com/opencollective/opencollective-api/blob/9255bd06ac4faae4740af36254595d4e7324dfd2/server/graphql/v2/object/VirtualCard.ts#L35-L50

> I'd be in favor of an `assignee` (Individual) argument to achieve this.

I've added a `assignee` argument and modified the front-end as well to pass that. Let me know if you see any issues. 🤔

Reply:
It's the assignee indeed. My feedback was really about the workaround using Account being too hacky.

Reply:
Edit: I misread the original issue and did not understand that we wanted to add this to the user settings.

The new API looks wrong too, it doesn't make sense to do:

```gql
collective(slug: "babel") {
  virtualCards(assignee: "betree) { ... }
}
```

and end up with virtual cards for Webpack assigned to betree in the list.

Reply:
Suggestion:

```gql
account(slug: "betree) {
  virtualCards(relation: ASSIGNED) { ... }
}
```

With `relation` set to `OWNED` by default.

Not sure whether `relation` is the best term, feel free to dig more.

Reply:
```graphql
collective(slug: "babel") { virtualCards(assignee: "betree) { ... } }
````

I'm fine with this API if we make sure that we only fetch cards assigned to "betree" in the "babel" collective. And not call cards assigned to "betree".

Reply:
If we don't add a virtual cards page in users settings (which I think is a good call) then we indeed don't need this `relation` thing and can directly go with an `assignee` arguments on `virtualCards`.

Reply:
Refactored the code to remove the virtual cards page in user's settings. So these changes have been reverted.